### ISSUE TITLE
type warnings on  RichProgressBar - > task_instance_completed -> parameter "results" vs "result"

### ISSUE THREAD

**Issue by dmzook** [InfrastructureAsCode-ch/nornir_rich#14]
It looks like nornir has added more type hints "Processor" causing a typing warning when trying to add the RichProgressBar processor to my nornir instance.
```
nr.with_processors(
            [RichProgressBar()]  
        )
```

```
Argument of type "list[RichProgressBar]" cannot be assigned to parameter "processors" of type "List[Processor]" in function "with_processors"
  "RichProgressBar" is incompatible with protocol "Processor"
    "task_instance_completed" is an incompatible type
      Type "(task: Task, host: Host, results: MultiResult) -> None" cannot be assigned to type "(task: Task, host: Host, result: MultiResult) -> None"
        Parameter name mismatch: "result" versus "results"
```

I tried a quick workaround that also pointed out the parameter name miss match in the task_instance_completed method.
```
from nornir.core.processor import Processor

class RichProgressBarTyped(RichProgressBar, Processor):
    pass
```

I propose updating RichProgressBar to inherit from Processor and update the parameter name 'results' to 'result' on method task_instance_completed'.

**Comment by ubaumann** [comment]
Thanks for opening the issue. What version of Nornir are you using?

**Comment by dmzook** [comment]
nornir            3.3.0 

It is the same issue in the latest version 3.4.1
https://github.com/nornir-automation/nornir/blob/d8c547cbcb377d365e53e5e0a7f96072659c6c52/nornir/core/processor.py#L33

**Comment by ubaumann** [comment]
I just released version v0.2.0. Let me know if this fixes the issue, and we can close it.
https://github.com/InfrastructureAsCode-ch/nornir_rich/releases/tag/v0.2.0

**Comment by dmzook** [comment]
I have confirm I see the issue is fixed.  Thank you.
